doc/sailfish-webview-qmozcontext.cpp

```diff
@@ -0,0 +1,380 @@
+
+    The \c removeObservers method should be used to unregister interest in multiple
+    topics simultaneously. The \a aObserversList should be set to contain a list
+    of all the topics that are no longer of interest.
+
+    Only topics that have previously been registered using \l addObserver or
+    \l addObservers should be included in the \a aObserversList.
+
+    This is equivalent to calling \l removeObserver multiple times.
+
+    See \l addObserver for more detailed info about notifications and observers.
+
+    \sa addObservers, removeObserver, removeObservers, recvObserve, notifyObservers
+*/
+void QMozContext::removeObservers(const std::vector<std::string> &aObserversList);
+
+/*!
+    \brief Sends a broadcast notification that can be observed by other components.
+
+    Sends a notification that will be picked up by any component that has
+    registered an observer for \a topic.
+
+    Additional data can be sent with the notification using the \a value
+    parameter encoded as a JSON string.
+
+    \sa addObserver, recvObserve
+*/
+void QMozContext::notifyObservers(const QString &topic, const QString &value);
+
+/*!
+    \brief Sends a broadcast notification that can be observed by other components.
+
+    Sends a notification that will be picked up by any component that has
+    registered an observer for \a topic.
+
+    Additional data can be sent with the notification using the \a value
+    parameter which should contain a structure (e.g. a \c{QVariantMap}) that will
+    be encoded as a JSON object.
+
+    \sa addObserver, recvObserve
+*/
+void QMozContext::notifyObservers(const QString &topic, const QVariant &value);
+
+/*!
+    \brief Starts the engine event loop.
+
+    When WebEngine is instantiated as a QML component this is called
+    automatically during initialization. However, it can be called earlier to
+    set the WebEngine event loop running in case the initialization process is
+    being overridden.
+
+    See \l {SailfishOS::WebEngine::initialize}{WebEngine::initialize} for more info.
+
+    The \a aDelay allows for an asynchronous startup. Calling without the
+    \a aDelay parameter specified will cause execution to happen on the nested
+    main loop, which will block until \l stopEmbedding is called.
+
+    \sa {SailfishOS::WebEngine::initialize}{WebEngine::initialize}, stopEmbedding, setProfile
+*/
+void QMozContext::runEmbedding(int aDelay = -1);
+
+/*!
+    \brief Stops the engine event loop.
+
+    Causes execution of the WebEngine to halt. If \l runEmbedding was called
+    without an \c aDelay parameter, this will cause the \l runEmbedding call to
+    unblock.
+
+    \sa {SailfishOS::WebEngine::initialize}{WebEngine::initialize}, runEmbedding, setProfile
+*/
+void QMozContext::stopEmbedding();
+
+/*!
+    \internal
+    \brief Sends the "final-ui-startup" notification to observers
+
+    For internal use? I don't actually see it used anywhere.
```

Comment:
I vaguely remember that this will be called by some xre function inside gecko upon first startup. But definitely this should be kept internal.

Reply:
I've removed the additional comment and left this with the brief text and marked as `\internal`.